endpoints/citrus-camel/src/main/java/org/citrusframework/camel/actions/CamelRunIntegrationAction.java

```diff
@@ -128,7 +131,24 @@ public void doExecute(TestContext context) {
                 throw new CitrusRuntimeException(String.format("Failed to start Camel integration - exit code %s", pao.getProcess().exitValue()));
             }
 
+            // Wait for first log to unsure the run process is correctly launched before setting the pid in context
+            int maxAttempts = CamelSettings.getMaxAttempts();
+            long delayBetweenAttempts = CamelSettings.getDelayBetweenAttempts();
+            for (int i = 0; i < maxAttempts; i++) {
+                String log = pao.getOutput();
+                if (log.length() > 0) {
+                    break;
+                }
+
+                try {
+                    Thread.sleep(delayBetweenAttempts);
+                } catch (InterruptedException e) {
+                    logger.warn("Interrupted while waiting for Camel integration '%s' first log".formatted(name), e);
```

Comment:
```suggestion
                    logger.warn("Interrupted while waiting for Camel integration '{}' first log", name, e);
```

please use the `logger`-internal formatting.